src/CMakeLists.txt

```diff
@@ -11,19 +11,44 @@ if(NOT CCPP_SUITES)
   set(CCPP_SUITES "FV3_CPT_v0,FV3_GFS_2017_gfdlmp,FV3_GFS_2017_gfdlmp_regional,FV3_GSD_SAR,FV3_GSD_v0,FV3_GFS_v15p2,FV3_GFS_v16,FV3_RRFS_v1beta,FV3_HRRR,FV3_RRFS_v1alpha")
 endif()
 
+if(NOT APP)
+  set(APP "ATM")
+endif()
+
+if(NOT CMAKE_BUILD_TYPE)
+  set(CMAKE_BUILD_TYPE "RELEASE")
+endif()
+
+list(APPEND UFS_WEATHER_MODEL_ARGS
+  "-DCMAKE_INSTALL_PREFIX=${CMAKE_INSTALL_PREFIX}"
+  "-DCCPP_SUITES=${CCPP_SUITES}"
+  "-DCMAKE_C_COMPILER=${MPI_C_COMPILER}"
+  "-DCMAKE_CXX_COMPILER=${MPI_CXX_COMPILER}"
+  "-DCMAKE_Fortran_COMPILER=${MPI_Fortran_COMPILER}"
+  "-DCMAKE_BUILD_TYPE=${CMAKE_BUILD_TYPE}"
+  "-DNETCDF_DIR=$ENV{NETCDF}"
+  "-D32BIT=ON"
```

Comment:
I like this modification to allow command-line arguments to CMake! While you're adding this, could we get the ability to have D32BIT and DINLINE_POST also be controlled by flags passed to CMake (similar to how you handled DATM)?

Reply:
@mkavulich I can add any options you'd like. Just verify what you'd like the user interface to look like
--32bit=<ON|OFF>?
--inline-post=<ON|OFF>?

Just an FYI the --component=<COMPONENT1,COMPONENT2,...> just takes the transforms this list into -DCOMPONENT1=ON -DCOMPONENT2=ON, do you want me to change this to --enable-option=<ITEM1,ITEM2,...> and --disable-option=<ITEM1,ITEM2,...> (disable macro would set -DITEM1=OFF -DITEM2=OFF. Doing so would cover all weather model options.

Reply:
I added --enable-options and --disable-options